### DISCUSSION TITLE
Checkout Action does not create local master and has no options for Tags breaking GitVersion

### DISCUSSION THREAD

**Discussion by giuliov** [community/community#25837] [Actions]
<p>The <strong>checkout</strong> action cannot be configured so that it ensure the presence of a local master branch.</p>
<p>Also it forces the --no-tags option, while there should be the possibility to get a repo tags.</p>
<p>Without these two (master branch and tags) the notable <a href="https://github.com/GitTools/GitVersion" rel="nofollow noopener">GitVersion</a> tool, used to version artifacts, does not work.<br>
If I explicitly add to the workflow</p>
<p><span>    - </span><span>uses</span><span>: </span><span>actions/checkout@v1</span><br>
<span>    - </span><span>run</span><span>: |</span><br>
<span>        </span><span>git fetch --tags</span><br>
<span>        </span><span>git branch --create-reflog master origin/master</span></p>
<p>then it works.</p>
<p>The expected behaviour would be setting an additional parameter at the checkout action, e.g.</p>
<p><span>    - </span><span>uses</span><span>: </span><span>actions/checkout@v1</span><br>
<span>      </span><span>with:</span></p>
<p><span>        </span><span>tags: true<br></span></p>
<p>and the master branch should always be available.<span><br></span></p>

**Comment by chrispat** [comment, marked as answer]
<p>We are working on an update that will enable better scripting of git as part of your worklfows.  You can see the initail work in the 1.1 version <a href="https://github.com/actions/checkout/pull/56" rel="nofollow noopener">https://github.com/actions/checkout/pull/56</a>.  And there will be a 2.0 version coming that will make sure you are not in a detached head state.</p>
